plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -315,24 +312,25 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
      * Labels that don't match for a particular resource type will be
      * rejected by the API.
      *
-     * For a list of valid labels, see:
+     * For a list of valid labels per resource, see:
      * https://cloud.google.com/logging/docs/api/v2/resource-list
     */
     mk_list_foreach(labels, &ins->properties) {
         kv = mk_list_entry(labels, struct flb_kv, _head);
-        if (strncasecmp(kv->key, rlabel_prefix, rlabel_prefix_len) == 0 &&
+        if (flb_sds_cmp(kv->key, rlabel_prefix, rlabel_prefix_len) == 0 &&
```

Comment:
So you're making the label prefix case-sensitive as well?

Reply:
Actually, `flb_sds_cmp` will not do a prefix comparison, so I'd suggest going back to `strncmp` and adding a comment.

Reply:
ahh the case, I missed that. I've added a case insensitive version flb_sds_casecmp() here e671b5ae

Reply:
The issue is that it checks for equivalent length, which the prefix will always be shorter so just using strncmp would be better in this case.

Reply:
sds cmp check for lengths, consider a best practice that "if I am comparing against a SDS, use sds cmp family: https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_sds.h#L92

Reply:
That only works if you are checking is the whole strings are the same. Since we are doing prefix checking here, it doesn't work.